diff --git a/cmake/boost.cmake b/cmake/boost.cmake
index c979055cf46..30eaa8534c9 100644
--- a/cmake/boost.cmake
+++ b/cmake/boost.cmake
@@ -41,7 +41,7 @@
 SET(BOOST_PACKAGE_NAME "boost_1_77_0")
 SET(BOOST_TARBALL "${BOOST_PACKAGE_NAME}.tar.bz2")
 SET(BOOST_DOWNLOAD_URL
-  "https://boostorg.jfrog.io/artifactory/main/release/1.77.0/source/${BOOST_TARBALL}"
+  "https://github.com/shogo82148/boost-mirror/releases/download/1.77.0/${BOOST_TARBALL}"
   )
 
 SET(OLD_PACKAGE_NAMES
