diff --git a/cmake/pcre.cmake b/cmake/pcre.cmake
index 3d4f163fab9..ceff3106e7d 100644
--- a/cmake/pcre.cmake
+++ b/cmake/pcre.cmake
@@ -43,7 +43,7 @@ MACRO(BUNDLE_PCRE2)
   ExternalProject_Add(
     pcre2
     PREFIX   "${dir}"
-    URL      "http://ftp.pcre.org/pub/pcre/pcre2-10.37.zip"
+    URL      "https://github.com/PhilipHazel/pcre2/releases/download/pcre2-10.37/pcre2-10.37.zip"
     URL_MD5  8c1699a725d4b28410adf4b964ebbcb7
     INSTALL_COMMAND ""
     CMAKE_ARGS
