diff --git a/storage/rocksdb/rocksdb/CMakeLists.txt b/storage/rocksdb/rocksdb/CMakeLists.txt
index aeef92e25..726e3755b 100644
--- a/storage/rocksdb/rocksdb/CMakeLists.txt
+++ b/storage/rocksdb/rocksdb/CMakeLists.txt
@@ -135,11 +135,11 @@ else()
     list(APPEND THIRDPARTY_LIBS ${BZIP2_LIBRARIES})
   endif()
 
-  if(WITH_LZ4)
-    find_package(lz4 REQUIRED)
-    add_definitions(-DLZ4)
-    list(APPEND THIRDPARTY_LIBS lz4::lz4)
-  endif()
+  # if(WITH_LZ4)
+  #   find_package(lz4 REQUIRED)
+  #   add_definitions(-DLZ4)
+  #   list(APPEND THIRDPARTY_LIBS lz4::lz4)
+  # endif()
 
   if(WITH_ZSTD)
     find_package(zstd REQUIRED)
