diff --git a/lplugin/auth_pam/mapper/pam_user_map.c b/plugin/auth_pam/mapper/pam_user_map.c
index b7bbbad8c..357b8ac08 100644
--- a/plugin/auth_pam/mapper/pam_user_map.c
+++ b/plugin/auth_pam/mapper/pam_user_map.c
@@ -39,7 +39,11 @@ and usually end up in /var/log/secure file.
 #include <grp.h>
 #include <pwd.h>
 
+#ifdef __FreeBSD__
+#include <security/pam_appl.h>
+#elif
 #include <security/pam_ext.h>
+#endif
 #include <security/pam_modules.h>
 
 #define FILENAME "/etc/security/user_map.conf"
