diff --git a/storage/rocksdb/rocksdb/util/autovector.h b/storage/rocksdb/rocksdb/util/autovector.h
index 5843fa8a1..cea6877e0 100644
--- a/storage/rocksdb/rocksdb/util/autovector.h
+++ b/storage/rocksdb/rocksdb/util/autovector.h
@@ -120,26 +120,19 @@ class autovector {
     }
 
     // -- Reference
-    reference operator*() {
+    reference operator*() const {
       assert(vect_->size() >= index_);
       return (*vect_)[index_];
     }
 
-    const_reference operator*() const {
-      assert(vect_->size() >= index_);
-      return (*vect_)[index_];
-    }
-
-    pointer operator->() {
-      assert(vect_->size() >= index_);
-      return &(*vect_)[index_];
-    }
-
     const_pointer operator->() const {
       assert(vect_->size() >= index_);
       return &(*vect_)[index_];
     }
 
+    reference operator[](difference_type len) const {
+      return *(*this + len);
+    }
 
     // -- Logical Operators
     bool operator==(const self_type& other) const {
