diff --git a/mysys/my_init.c b/mysys/my_init.c
index 138b4697f97..ae9461d14d0 100644
--- a/mysys/my_init.c
+++ b/mysys/my_init.c
@@ -178,24 +178,7 @@ my_bool my_init(void)
   my_progname_short= "unknown";
   if (my_progname)
   {
-    char link_name[FN_REFLEN];
     my_progname_short= my_progname + dirname_length(my_progname);
-    /*
-      if my_progname_short doesn't start from "mariadb", but it's
-      a symlink to an actual executable, that does - warn the user.
-      First try to find the actual name via /proc, but if it's unmounted
-      (which it usually is on FreeBSD) resort to my_progname
-    */
-    if (strncmp(my_progname_short, "mariadb", 7))
-    {
-      int res= 1;
-#ifdef EXE_LINKPATH
-      res= my_readlink(link_name, EXE_LINKPATH, MYF(0));
-#endif
-      if ((res == 0 || my_readlink(link_name, my_progname, MYF(0)) == 0) &&
-           strncmp(link_name + dirname_length(link_name), "mariadb", 7) == 0)
-      my_error(EE_NAME_DEPRECATED, MYF(MY_WME), link_name);
-    }
   }
 
   /* Initialize our mutex handling */
