diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index dbef87c409f..666de3cdbe5 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -146,6 +146,10 @@ IF(MSVC)
   #TODO: update the code and remove the disabled warnings
   SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4800 /wd4805 /wd4996")
   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4800 /wd4805 /wd4996 /we4099")
+
+  # use C11 and C++14 standard
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /std:c11")
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++14")
 ENDIF()
 
 # Always link with socket library
