diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index dbef87c409f..219b8637b98 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -125,7 +125,23 @@ IF(MSVC)
     ENDIF()
     SET("${flag}" "${${flag}} /EHsc")
   ENDFOREACH()
-  
+
+  # use C11
+  FOREACH(flag 
+      CMAKE_C_FLAGS_MINSIZEREL
+      CMAKE_C_FLAGS_RELEASE    CMAKE_C_FLAGS_RELWITHDEBINFO 
+      CMAKE_C_FLAGS_DEBUG      CMAKE_C_FLAGS_DEBUG_INIT)
+    SET("${flag}" "${${flag}} /std:c11")
+  ENDFOREACH()
+
+  # use C++14
+  FOREACH(flag 
+      CMAKE_CXX_FLAGS_MINSIZEREL
+      CMAKE_CXX_FLAGS_RELEASE  CMAKE_CXX_FLAGS_RELWITHDEBINFO
+      CMAKE_CXX_FLAGS_DEBUG    CMAKE_CXX_FLAGS_DEBUG_INIT)
+    SET("${flag}" "${${flag}} /std:c++14")
+  ENDFOREACH()
+
   FOREACH(type EXE SHARED MODULE)
     FOREACH(config DEBUG RELWITHDEBINFO RELEASE MINSIZEREL)
       SET(flag "CMAKE_${type}_LINKER_FLAGS_${config}")
