diff --git a/sql/locks/shared_spin_lock.cc b/sql/locks/shared_spin_lock.cc
index 7cbb948e0ac..27a2e7116f3 100644
--- a/sql/locks/shared_spin_lock.cc
+++ b/sql/locks/shared_spin_lock.cc
@@ -239,7 +239,7 @@ lock::Shared_spin_lock &lock::Shared_spin_lock::try_or_spin_exclusive_lock(
   {
     this->spin_exclusive_lock();
   }
-  my_atomic_store64(&this->m_exclusive_owner, self);
+  my_atomic_store64(&this->m_exclusive_owner, int64(self));
   return (*this);
 }
 
